compiler/optimizer/VPHandlers.cpp

```diff
@@ -5704,6 +5704,13 @@ TR::Node *constrainAcall(TR::ValuePropagation *vp, TR::Node *node)
             TR::VPResolvedClass *newTypeConstraint = NULL;
             if (constraint)
                {
+               // Do nothing if the class of the object doesn't implement Cloneable
+               if (constraint->getClass() && !vp->comp()->fej9()->isCloneable(constraint->getClass()))
+                  {
+                  if (vp->trace())
+                     traceMsg(vp->comp(), "Object Clone: Class constraint on node %p doesn't exist or the class is not cloneable, quit\n", node);
+                  return node;
```

Comment:
I don't think we want to return early here since it will mess up the hit/miss debug counters below. This check probably needs to go down around the code where we add the clone to the transformation list and we should probably move the debug counter logic down there so that the counters count whether we hit or missed on adding it to the transformation list.